internal/querycoordv2/task/merger.go

```diff
@@ -73,40 +68,23 @@ func (merger *Merger[K, R]) schedule(ctx context.Context) {
 	go func() {
 		defer merger.wg.Done()
 		ticker := time.NewTicker(500 * time.Millisecond)
+		defer ticker.Stop()
 		for {
 			select {
 			case <-ctx.Done():
+				close(merger.outCh)
 				log.Info("Merger stopped due to context canceled")
 				return
 
 			case <-merger.stopCh:
+				close(merger.outCh)
```

Comment:
Shall we need to process leftover tasks within the queues before stopping?

Reply:
Since it is closed, so we just left because querynode might shut down at the time

Reply:
it is like a graceful stop, all tasks need to retry if force stop